suite-common/wallet-graph/src/graphThunks.ts

```diff
@@ -0,0 +1,150 @@
+import { selectAccountsByNetworkSymbols } from '@suite-common/wallet-core';
+
+import { LineGraphTimeFrameValues } from './types';
+import { getLineGraphAllTimeStepInMinutes } from './graphUtils';
+import { timeSwitchItems } from './config';
+import { actionPrefix } from './graphActions';
+
+const getOldestAccountBalanceMovementTimestamp = async (accounts: Account[]) => {
+    const promises = accounts.map(async account => {
+        const response = await TrezorConnect.blockchainGetAccountBalanceHistory({
+            coin: account.symbol,
+            descriptor: account.descriptor,
+            groupBy: 3600 * 24, // day
+        });
+        try {
+            if (response?.success) {
+                const { payload } = response;
+                return payload.map(balanceGroup => balanceGroup.time);
+            }
+        } catch (error) {
+            // eslint-disable-next-line no-console
+            console.log(error);
+        }
+        return [];
+    });
+    const accountBalanceTimestamps = await Promise.all(promises);
+    return Math.min(...accountBalanceTimestamps.flat());
+};
+
+const getValueBackInMinutes = async (
+    accounts: Account[],
+    timeFrame: LineGraphTimeFrameValues,
+): Promise<number> => {
+    const { valueBackInMinutes } = timeSwitchItems[timeFrame];
+    if (valueBackInMinutes) {
+        return valueBackInMinutes;
+    }
+    const oldestAccountBalanceChangeUnixTime = await getOldestAccountBalanceMovementTimestamp(
+        accounts,
+    );
+    return differenceInMinutes(new Date(), new Date(oldestAccountBalanceChangeUnixTime * 1000));
+};
+
+const fetchAccountsGraphData = async (
+    accounts: Account[],
+    { from, to, groupBy }: { from: number; to: number; groupBy: number },
+) => {
+    const promises = accounts.map(async account => {
+        const response = await TrezorConnect.blockchainGetAccountBalanceHistory({
+            coin: account.symbol,
+            descriptor: account.descriptor,
+            from,
+            to,
+            groupBy,
+        });
+        try {
+            if (response?.success) {
+                // TODO
+            }
+        } catch (error) {
+            // eslint-disable-next-line no-console
+            console.log(error);
+        }
+    });
+    const groupedAccountBalance = await Promise.all(promises);
+    // TODO recalculate historic rates etc.
+    return groupedAccountBalance;
+};
+
+export const getGraphPointsForAccountsThunk = createThunk(
+    `${actionPrefix}/getGraphPointsForAccountsThunk`,
+    async (
+        {
+            section,
+            timeFrame,
+        }: {
+            section: 'dashboard' | 'account';
+            timeFrame: LineGraphTimeFrameValues;
+        },
+        { getState },
+    ) => {
+        const endOfRangeDate = new Date();
+
+        // FIXME mobile app currently supports only btc so it is hardcoded for now
+        const accounts = selectAccountsByNetworkSymbols(getState(), ['btc']);
```

Comment:
And testnet maybe?

Reply:
We should definitely add some const with supported networks that we will use everywhere. Not we have this hardcoded in quite a lot places

Reply:
You both are right... lets do this as soon as possible because there are a lot of similar TODO comments like this one in the app. It is starting to be messy.